src/monexamine.cpp

```diff
@@ -313,6 +321,60 @@ bool monexamine::mech_hack( monster &z )
     return false;
 }
 
+static int prompt_for_amount( const char *const msg, const int max )
+{
+    const std::string formatted = string_format( msg, max );
+    const int amount = string_input_popup()
+                       .title( formatted )
+                       .width( 20 )
+                       .text( to_string( max ) )
+                       .only_digits( true )
+                       .query_int();
+
+    return clamp( amount, 0, max );
+}
+
+bool monexamine::pay_bot( monster &z )
+{
+    time_duration friend_time = z.get_effect_dur( effect_pet );
+    const int card_count = g->u.amount_of( "cash_card" );
+    const int charge_count = card_count ? g->u.charges_of( "cash_card" ) : 0;
```

Comment:
You can call `charges_of` unconditionally, you don't need to check whether the player has any cash cards at all. If they have non, the charges count will be 0.
```suggestion
    const int charge_count = g->u.charges_of( "cash_card" );
```